experiments/compositions/composition/api/v1alpha1/composition_types.go

```diff
@@ -45,57 +43,71 @@ type ResourceRef struct {
 	// resource: ServiceIdentity.serviceusage.cnrm.cloud.google.com/v1beta1//sqladmin.googleapis.com
 
 	// OPTION 2
-	Group      string `json:"group,omitempty" protobuf:"bytes,1,name=group"`
-	Version    string `json:"version,omitempty" protobuf:"bytes,2,opt,name=version"`
-	Kind       string `json:"kind" protobuf:"bytes,3,name=kind"`
-	Name       string `json:"name,omitempty" protobuf:"bytes,4,name=name"`
-	NameSuffix string `json:"nameSuffix,omitempty" protobuf:"bytes,2,name=nameSuffix"`
+	Group      string `json:"group,omitempty"`
+	Version    string `json:"version,omitempty"`
+	Kind       string `json:"kind"`
+	Name       string `json:"name,omitempty"`
+	NameSuffix string `json:"nameSuffix,omitempty"`
 }
 
 type FieldRef struct {
-	Path string `json:"path" protobuf:"bytes,1,name=path"`
-	As   string `json:"as" protobuf:"bytes,2,name=as"`
+	Path string `json:"path"`
+	As   string `json:"as"`
 }
 
 type ValuesFrom struct {
-	Name        string      `json:"name" protobuf:"bytes,1,name=name"`
-	ResourceRef ResourceRef `json:"resourceRef" protobuf:"resourceRef,2,name=resourceref"`
-	FieldRef    []FieldRef  `json:"fieldRef" protobuf:"fieldRef,3,name=fieldref"`
+	Name        string      `json:"name"`
+	ResourceRef ResourceRef `json:"resourceRef"`
+	FieldRef    []FieldRef  `json:"fieldRef"`
+}
+
+type Jinja2 struct {
+	Template string `json:"template"`
+}
+
+// ConfigReference - For BYO Expanders, we can extend it
+type ConfigReference struct {
+	APIGroup  string `json:"APIGroup"`
+	Name      string `json:"name"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type ExpanderConfig struct {
+	// Built in expanders
+	Jinja2 *Jinja2 `json:"jinja2,omitempty"`
+	// Soon..
+	// Getter *Getter `json:"getter,omitempty"`
```

Comment:
When we get to the "Getter" I would like us to take a look at how to make this more generic. It seems like this should be a reference to a config object appropriate to the expander and we should have 1 such field for all the expander types rather than for each expander type.

Reply:
The disadvantage with this approach is the composition is not inline. Meaning when defining a composition we would need to define a few expander objects and reference it in the composition. Is this the experience we want for built in expanders as well ?

Reply:
Adding a comment and a TODO to not have inline expanders long run